mcs/class/corlib/System.Reflection.Emit/ModuleBuilder.cs

```diff
@@ -107,18 +109,35 @@ public class ModuleBuilder : Module, _ModuleBuilder {
 
 			if (emitSymbolInfo) {
 				Assembly asm = Assembly.LoadWithPartialName ("Mono.CompilerServices.SymbolWriter");
-				if (asm == null)
-					throw new TypeLoadException ("The assembly for default symbol writer cannot be loaded");
+				if (asm == null) {
+					WarnAboutSymbolWriter ("Failed to load the default Mono.CompilerServices.SymbolWriter assembly");
+					return;
+				}
 
 				Type t = asm.GetType ("Mono.CompilerServices.SymbolWriter.SymbolWriterImpl", true);
```

Comment:
Better fix is not to change Activator code but change GetType not to throw and check for return value not to be null